src/clincoded/static/components/variant_central/interpretation/segregation/evidenceTable.js

```diff
@@ -0,0 +1,478 @@
+                        else if (col === 'comments') {
+                            // Display the comment for current criteria in this column
+                            let commentCol = criteriaName + '_comment';
+                            nodeContent = null;
+                            if (commentCol in obj) {
+                                nodeContent = obj[commentCol];
+                            }
+                            node = <td className='word-break-all' key={`cell_${i++}`}>
+                                {nodeContent}
+                            </td>
+                        } else {
+                            // Display value for other columns
+                            node = <td key={`cell_${i++}`} rowSpan={subRows}>
+                                {nodeContent}
+                            </td>
+                        }
+                        inner.push(node);
+                    });
+                }
+                // Add the edit/delete buttons if user can modify this evidence
+                this.addButtons(i++, row, inner, subRows);
+                let outer = <tr key={`row_${i++}`}>{inner}</tr>
+                rows.push(outer);
+
+                // Add other criteria rows if more is available for this evidence row.
+                otherCriteria.forEach(criteria => {
+                    outer = this.addAnotherCriteriaRow(criteria, colNames, obj, i);
+                    rows.push(outer);
+                    // Hack for unique key
+                    i = i + 10;
+                });
+            }
+        });
+        return rows;
+    },
+
+    /**
+     * Set the evidence table headers
+     */
+    tableHeader() {
+        let cols = this.state.tableFormat.cols.map(col => {
+            let criteriaCodes = this.getCriteriaCodes(col.key);
+            if (criteriaCodes.length > 0) {
+                criteriaCodes = criteriaCodes[0].codes;
+                return <th key={col.key}>{`${col.title} [${criteriaCodes.join(',')}]`}</th>
+            }
+            return <th key={col.key}>{col.title}</th>;
+        });
+        cols.push(<th key="editDelete"></th>)
+        return cols;
+    },
+
+    /**
+     * Check if there is evidence to be displayed for this panel.
+     */
+    hasTableData() {
+        // relevant columns non empty -> return true
+        // relevant columns empty -> return False
+        let relevantData = _.find(extraEvidence.sheetToTableMapping, o => o.subcategory === this.props.subcategory);
+        let cols = relevantData.cols.map(o => o.key);
+        let foundData = false;
+        this.props.tableData.forEach(row => {
+            if (row.source && row.source.data) {
+                let obj = row.source.data;
+                if (foundData) {
+                    return;
+                }
+                cols.forEach(col => {
+                    if (obj[col] && obj[col] != '') {
+                        foundData = true;
+                        return;
```

Comment:
`Return` does not break out of the forEach. You can use `cols.some()` and change this line to `return true`